File: Data Structures & Algorithms/valid-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

# get maxvalue of left subtree
# get minvalue of right subtree
# if root node is greater than maxvalue and smaller than minValue, BST is valid
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        def valid(node, left, right):
            if not node:
                return True
            
            if not (node.val > left and node.val < right):
                return False

            return valid(node.left, left, node.val) and valid(node.right, node.val, right)
        
        return valid(root, float("-inf"), float("inf"))

File: Data Structures & Algorithms/sqrtx/submission-0.kt
class Solution {
    fun mySqrt(x: Int): Int {
        var left = 0
        var right = x
        
        while(left <= right){
            val mid = left + (right - left) / 2
            val value = mid.toLong() * mid

            if(x > value){
                left = mid + 1
            } else if (x < value){
                right = mid - 1
            } else {
                return mid
            }
        }

        return right
    }
}
